src/util.cpp

```diff
@@ -952,3 +954,51 @@ int GetNumCores()
 #endif
 }
 
+VersionInfo::VersionInfo(uint32_t version)
+{
+    if((version >> 24) > 0)
+        throw std::runtime_error("Invalid version format");
+    if(version == 0)
+        throw std::runtime_error("Invalid version format");
+    nVersion = version;
+}
+
+VersionInfo::VersionInfo(const std::string& versionInfo)
+{
+    std::vector<std::string> tokens;
+    boost::split(tokens, versionInfo, boost::is_any_of("."));
+    if(tokens.size() != 3)
+        throw std::runtime_error("Invalid version format");
+    uint32_t version = 0;
+    for(unsigned idx = 0; idx < 3; idx++)
+    {
+        if(tokens[idx].length() == 0)
+            throw std::runtime_error("Invalid version format");
+        if(!std::none_of(tokens[idx].begin(), tokens[idx].end(),
```

Comment:
Looks like this will allow non-digit characters in the version string. Is this intentional ?